app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupMakingSettingsViewHolder.java

```diff
@@ -75,35 +82,64 @@ private void applyNumberOfGroups() {
+        }
+    }
+
     @OnTextChanged(value = R.id.num_of_names_per_group, callback = OnTextChanged.Callback.AFTER_TEXT_CHANGED)
     void afterNamesPerGroupChanged() {
-        String namesPerGroupInput = namesPerGroup.getText().toString().trim();
-        boolean isInputValid = isInputValid(namesPerGroupInput);
-        if (namesPerGroup.isFocused() && isInputValid) {
+        namesPerGroupInput = namesPerGroup.getText().toString().trim();
+        boolean isInputValid = isInputValid();
+        if (namesPerGroup.isFocused() && isInputValid && autocomplete.isChecked()) {
             numGroups.setText(getMatchingNumber(namesPerGroupInput));
         }
         dialog.getActionButton(DialogAction.POSITIVE).setEnabled(isInputValid);
+        maybeShowUnevenGroupsWarning();
     }
 
     @OnTextChanged(value = R.id.num_of_groups, callback = OnTextChanged.Callback.AFTER_TEXT_CHANGED)
     void afterGroupNumChanged() {
-        String numberOfGroupsInput = numGroups.getText().toString().trim();
-        boolean isInputValid = isInputValid(numberOfGroupsInput);
-        if (numGroups.isFocused() && isInputValid) {
+        numberOfGroupsInput = numGroups.getText().toString().trim();
+        boolean isInputValid = isInputValid();
+        if (numGroups.isFocused() && isInputValid && autocomplete.isChecked()) {
             namesPerGroup.setText(getMatchingNumber(numberOfGroupsInput));
         }
         dialog.getActionButton(DialogAction.POSITIVE).setEnabled(isInputValid);
+        maybeShowUnevenGroupsWarning();
     }
 
     // Given the number of names per group or number of groups, returns the corresponding number to use all the names
     private String getMatchingNumber(String inputtedNumber) {
         int newNumber = Integer.parseInt(inputtedNumber);
-        int offset = (int) Math.ceil((double) settings.getNameListSize() / newNumber);
-        return String.valueOf(offset);
+        double offset = (double) settings.getNameListSize() / newNumber;
+        int matchingNumber = (int) Math.ceil(offset);
+        return String.valueOf(matchingNumber);
+    }
+
+    // show warning if the groups don't have the same names quantity
+    private void maybeShowUnevenGroupsWarning() {
+        if (!isInputValid()) {
+            return;
+        }
+        int namesNumber = Integer.parseInt(namesPerGroupInput);
+        int groupsNumber = Integer.parseInt(numberOfGroupsInput);
+        int totalNumber = namesNumber * groupsNumber;
+        if (totalNumber > settings.getNameListSize()) {
+            warningMessage.setVisibility(View.VISIBLE);
+        } else {
+            warningMessage.setVisibility(View.GONE);
+        }
     }
 
     // Returns true if the inputted text is a positive number
-    private boolean isInputValid(String inputText) {
-        return inputText.length() > 0 && Integer.parseInt(inputText) > 0;
+    private boolean isInputValid() {
+        return (namesPerGroupInput != null && numberOfGroupsInput != null)
```

Comment:
We can use TextUtils.isEmpty() to check for both empty AND null. It's a nice little utility method built into Android :)